src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,113 @@
 import { useState } from 'react';
+import { Movie } from '../../types/Movie';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface MovieForm {
+  title: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+  description: string;
+}
+
+export const NewMovie = ({ onAdd }: { onAdd: (movie: Movie) => void }) => {
+  const [form, setForm] = useState<Movie>({
+    title: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+    description: '',
+  });
+
+  const [touched, setTouched] = useState<{ [key: string]: boolean }>({});
+
+  const handleBlur = (name: string) => {
+    setTouched(prev => ({ ...prev, [name]: true }));
+  };
+
+  const handleChange = (name: string, value: string) => {
+    setForm(prevForm => ({
+      ...prevForm,
+      [name]: value,
+    }));
+  };
+
+  const isFormValid = () => {
+    const requiredFields = ['title', 'imgUrl', 'imdbUrl', 'imdbId'];
+    return requiredFields.every(field => !!form[field as keyof MovieForm]);
+  };
```

Comment:
```suggestion
  const isFormValid = () => {
    const requiredFields = ['title', 'imgUrl', 'imdbUrl', 'imdbId'];
    return requiredFields.every(field => !!form[field as keyof MovieForm].trim());
  };
```
to fix my prev comment